### PR TITLE
feat(mfa): add new enroll mfa policy to automatically manage enrollment step

### PR DESCRIPTION
**Pull request by tcompiegne**
closes gravitee-io/issues#7753

### CODE REVIEW

File: gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/java/io/gravitee/am/policy/enroll/mfa/EnrollMfaPolicy.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                    case EMAIL:
+                        Map<String, Object> additionalData = Collections.singletonMap(FactorDataKeys.KEY_MOVING_FACTOR, generateInitialMovingFactor(user));
+                        enrolledFactor.setSecurity(new EnrolledFactorSecurity(SHARED_SECRET, SharedSecret.generate(), additionalData));
+                        enrolledFactor.setChannel(new EnrolledFactorChannel(EnrolledFactorChannel.Type.EMAIL, enrollmentValue));
+                        break;
+                    case HTTP:
+                        enrolledFactor.setChannel(new EnrolledFactorChannel(EnrolledFactorChannel.Type.HTTP, enrollmentValue));
+                        break;
+                    default:
+                        return Single.error(new IllegalStateException("Unexpected value: " + factor.getFactorType().getType()));
```
Comment (remisultan) [inline gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/java/io/gravitee/am/policy/enroll/mfa/EnrollMfaPolicy.java:162]: When the content of the EL is `null` we return an `Internal Server Error`.
Is it up to the user to make sure the expression language returns something ? Or should we redirect to the Enroll page to make sure the user isn't blocked ? We could just let the user continue through the flow and if he hasn't enroll he will presented with the Enroll page ?

WDYT 🤔  ?

image

**Review by remisultan — COMMENTED** [review]
Usage is fine 👍 
A few minor comments added

File: gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/java/io/gravitee/am/policy/enroll/mfa/EnrollMfaPolicy.java (hunk 1 above)
Comment (tcompiegne) [reply gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/java/io/gravitee/am/policy/enroll/mfa/EnrollMfaPolicy.java:164]: FP

File: gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/resources/plugin.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+id=policy-am-enroll-mfa
+name=Enroll MFA
+version=${project.version}
+description=${project.description}
+class=io.gravitee.am.policy.enroll.mfa.EnrollMfaPolicy
+type=policy
+category=others
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply gravitee-am-policy/gravitee-am-policy-enroll-mfa/src/main/resources/plugin.properties]: FP

**Review by remisultan — APPROVED** [review]

**Pull request merged**